lib/ansible_tower_client/job_template.rb

```diff
@@ -10,11 +10,15 @@ def initialize(api, raw_body)
       super
     end
 
-    def launch(vars = {})
+    def launch(options = {})
       launch_url = "#{url}launch/"
-      extra = JSONValues.new(vars).extra_vars
-      resp = api.post(launch_url, extra).body
-      job = JSON.parse(resp)
+      new_limit = options.dup.delete('limit')
+      response = with_temporary_changes(new_limit) do
+        api.post(launch_url, options).body
+      end
+
+      job = JSON.parse(response)
+      raise InvalidOptions.new(self, Error.parse(response)) if job['job'].nil?
```

Comment:
The thought was so we could see errors like this `AnsibleTowerClient::InvalidOptions: Error on variables_needed_to_start: ["'instance_ids' value missing", "'crap_id' value missing"]`